code/game/atoms_movable.dm

```diff
@@ -476,18 +476,30 @@
 	var/pixel_y_diff = 0
 
 	var/direction = get_dir(src, A)
-	if(direction & NORTH)
-		pixel_y_diff = 8
-	else if(direction & SOUTH)
-		pixel_y_diff = -8
-
-	if(direction & EAST)
-		pixel_x_diff = 8
-	else if(direction & WEST)
-		pixel_x_diff = -8
+	switch(direction)
```

Comment:
this is worse code, change it back.

Reply:
Why is it worse?

Reply:
because the old code treats them like bitflags instead of enums, and direction is bitflags not enums

Reply:
This is actually better performant, and doing bitflag comparisons adds nothing. It's neither more clear nor more efficient.